### PR TITLE
Add a way to find `.eh_frame_hdr` using `__GNU_EH_FRAME_HDR`.

### PR DESCRIPTION
**Pull request by sunfishcode**
GNU ld on at least some platforms adds a symbol `__GNU_EH_FRAME_HDR` for
the `.eh_frame_hdr` section. Add a cargo feature `fde-gnu-eh-frame-hdr`
to enable use of this.

### CODE REVIEW

File: src/unwinder/find_fde/mod.rs [hunk 1]
```diff
@@ -35,6 +37,10 @@ impl FDEFinder for GlobalFinder {
         if let Some(v) = fixed::get_finder().find_fde(pc) {
             return Some(v);
         }
+        #[cfg(feature = "fde-gnu-eh-frame-hdr")]
+        if let Some(v) = gnu_eh_frame_hdr::get_finder().find_fde(pc) {
+            return Some(v);
+        }
```
Comment (nbdd0121) [inline src/unwinder/find_fde/mod.rs]: Should this perhaps be moved up so that it's searched before PHDR?

**Review by nbdd0121 — COMMENTED** [review]
Thanks a lot! I was trying to find such a symbol previously but didn't find any documentations. `ld --verbose` does not contain such a symbol, but I could see `GNU_EH_FRAME_HDR` indeed exists with readelf. Do you know when/where is this symbol added?

File: src/unwinder/find_fde/mod.rs (hunk 1 above)
Comment (sunfishcode) [reply src/unwinder/find_fde/mod.rs]: Sure.

**Comment by sunfishcode** [comment]
[GNU ld (BFD) defines it](https://github.com/redox-os/binutils-gdb/blob/master/bfd/elf-eh-frame.c#L1723).

**Comment by sunfishcode** [comment]
I don't know of any documentation for this though; I only know of it because I found it while looking at a disassembly.

**Comment by nbdd0121** [comment]
Thanks a lot!

**Pull request merged**